independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java

```diff
@@ -1173,8 +1173,15 @@ protected void implementCreate(ClassOutput classOutput, ClassCreator beanCreator
                         paramTypesArray,
                         declaringProviderInstanceHandle, argsArray));
             } else {
-                create.assign(instanceHandle, create.invokeVirtualMethod(MethodDescriptor.of(producerMethod),
-                        declaringProviderInstanceHandle, referenceHandles));
+                ResultHandle invokeMethodHandle;
+                if (Modifier.isStatic(producerMethod.flags())) {
```

Comment:
For static producers we don't have to allocate the `declaringProviderInstanceHandle` at all. And we need to fix the reflection fallback for private methods too.